src/werkzeug/http.py

```diff
@@ -1195,7 +1195,7 @@ def _parse_pairs():
 
 def dump_cookie(
     key: str,
-    value: str = "",
+    value: t.Union[bytes, str] = "",
```

Comment:
Speaking of fixes to type hints, I just noticed this on line 1008:
```
    data: t.Optional[t.Union[bytes]] = None,
```
It doesn't make sense to use `Union` there unless some other type besides `bytes` is accepted (and from a quick look I don't think it is). Worth fixing while you're in here?

Reply:
Done.